eth-contracts/contracts/CKBChain.sol

```diff
@@ -65,10 +69,67 @@ contract CKBChain is ICKBChain, ICKBSpv {
 
     /// #ICKBChain
     function addHeaders(bytes calldata data) external {
+        // 1. view decode from data to headers view
+        bytes29 headerVecView = data.ref(uint40(ViewCKB.CKBTypes.HeaderVec));
+
+        // 2. iter headers
+        uint32 length = headerVecView.lengthHeaderVec();
+        uint32 index = 0;
+        while (index < length) {
+            bytes29 headerView = headerVecView.indexHeaderVec(index);
+            _addHeader(headerView);
+            index++;
+        }
+    }
+
+    function _addHeader(bytes29 headerView) private {
+        if (!verifyHeader(headerView)) {
```

Comment:
Is `require` better here?
I can not find any situation that we should support partial correct headers.

Reply:
My idea here is that if someone submits headers correctly(e.g. 5 correct headers), but only because someone submitted part of the headers(e.g. 2 headers) that they submitted was submitted first, then the rest of the headers(e.g. 3 headers) still has the ability to be stored on chain. Here, the main consideration is that the cost of gas is expensive.

（by the way, this part of the code was accidentally committed to this branch, so I just remove this part of the code in the pr）